### ISSUE TITLE
Leads/Opportunities should include Contact emails in the count

### ISSUE THREAD

**Issue by sadortun** [orocrm/crm#197]
Hello Oro Team,

We've just noticed that the  **Last contacted** field doesnt include the related entities. 

For example, if i send to a contact an email, the Account, Business Account and Opportunity will not be counted as part of their activities.

![image](https://cloud.githubusercontent.com/assets/794449/17163039/52176d96-5389-11e6-8c84-bc08dbd08010.png)

An other member of the team could look at this and think the customer has not been contacted yet. To find out if the customer has been contacted, you will have to dig into the account, and look in each of the contacts.

I am not sure if it's a bug or an intended behaviour, but it fail the folloing usability checks : 
1. Software should be easy to use and hard to misuse. 
2. Obey the principle of least astonishment. 

As a suggestion, i would recommend one of the following : 
- Rename the label for a more explicit text 
![image](https://cloud.githubusercontent.com/assets/794449/17163201/c421dd58-538a-11e6-9b52-4c73078c72cc.png)
- Change the way the  contaced field is calculated
- Dispaly the contacted activithy for this specific entity, and the related activity.
![image](https://cloud.githubusercontent.com/assets/794449/17163215/e0fce2ba-538a-11e6-97e9-9a14e44df611.png)

Have a good day,
Samuel

**Comment by artml** [comment]
@sadortun I'm not sure about this. You can contact your customer on many affairs, not necessarily this particular business opportunity. (Especially if there are two of them in the works simultaneously.)

> This is particulary true (and sad) when you already have started a discussion with a customer, and then create an Opportunity to track the evolution.
This will be partially fixed in 1.10 when Opportunities will inherit all the activities recorded towards a preceding Lead. I will double-check about the contact count though, thank you for pointing this out.

**Comment by artml** [comment]
Actually, our Context functionality should already solve this problem. If you had a prior conversation with the client about a perspective sale and created the opportunity later, just add the opportunity context to your email thread or a call log.

It would be really great to have it added automatically but this will require a lot of smartness on our side—to scan the conversations and guess if they can relate to this new opportunity in some way. This is doable, and will make a great feature, but it's not easy.

**Comment by sadortun** [comment]
I agree with you that detecting if a specific opportunity is discussed in a conversation is impossible. The more general point here  is that to make the **Last contacted** field usefull you have to add to the context all the related entities.

Another example:
I have an Opportunity with a related Contact and Business Customers. If i log the call on the opportunity, i think this activity should be visible on thee Business Account, and Account that are related to the Opportunity.

Currently, even if there is a lot of activity on one or many Opportunities, the Account and Business Customers dispalay "**Not Contacted Yet**", which seems wrong.

For example, searching for Business Customers 
![image](https://cloud.githubusercontent.com/assets/794449/17182447/8525826e-53f1-11e6-8d2f-1bbdd3109dbe.png)  return an empty dataset which is not accurate  since many Business Customers were contacted about various leads and opportunities.


As i mentioned above, if this is the intended behaviour, it might worth it to clarify the docs.

Thanks,
Samuel
